### PR TITLE
Registry perf3

### PR DESCRIPTION
**Pull request by ludoo0d0a**
finally my PR for search engine optimization

proper branch from last 1.6.0, and cherry pick + cleanup

(sorry again for bad quality)

### CODE REVIEW

File: lib/widgets/registry.dart [hunk 1]
```diff
@@ -6263,5 +6267,21 @@ class Registry {
...
+    buildIndexedStrings();
+
+  }
+
+  // Build indexed strings : concatenated lower case no accent
+  static void buildIndexedStrings() {
+    indexedTools = toolList.where((tool) {
+      var _indexedStrings = removeAccents(tool.searchStrings.join(' ').toLowerCase());
+      if (_indexedStrings == null || _indexedStrings.length == 0) return false;
+      tool.indexedStrings = _indexedStrings;
```
Comment (S-Man42) [inline lib/widgets/registry.dart:6283]: `tool.indexedStrings` is not defined.... You added private `_indexedStrings` to `GCWTool` instead. I guess this needs to be public instead?

File: lib/widgets/common/gcw_tool.dart [hunk 2]
```diff
@@ -124,6 +125,12 @@ class GCWTool extends StatefulWidget {
     }
   }
 
+  String get indexedStrings => _indexedStrings;
```
Comment (S-Man42) [inline lib/widgets/common/gcw_tool.dart:128]: Why not simply make _indexedStrings (and remove explicit getter and setter) to indexedStrings? AFAIK this results in the same behaviour. Am I wrong?

**Comment by S-Man42** [comment]
Some minor things (see comments). Looks good so far nonetheless: Great work! :)

**Comment by S-Man42** [comment]
One enhancement comes to my mind: Remove duplicate values: If, for example, there's a word used in two searchstring files (maybe COMMON and ENGLISH or theres the same word in two languages), you don't need to add both... So the best would be to collect all string in a Set (=> removes duplicates) and convert this to a List afterwards

File: lib/widgets/main_view.dart [hunk 3]
```diff
@@ -462,27 +464,21 @@ class _MainViewState extends State<MainView> {
...
   List<GCWTool> _getSearchedList() {
-    var list = Registry.toolList;
-    String searchstring = '';
-
-    list = list.where((tool) {
-      searchstring = tool.searchStrings.join(' ').toLowerCase();
-      if (searchstring == null || searchstring.length == 0) return false;
-
-      var found = true;
+    Stopwatch stopwatch = Stopwatch()..start();
```
[the remark this answers is not in the document]
Comment (ludoo0d0a) [reply lib/widgets/main_view.dart:467]: I waited the last moment. I wasn't sure.
ok I will remove it then

File: lib/widgets/common/gcw_tool.dart (hunk 2 above)
Comment (ludoo0d0a) [reply lib/widgets/common/gcw_tool.dart:128]: getter only, no setter 
but yes, more or less the same result

Comment (ludoo0d0a) [reply lib/widgets/common/gcw_tool.dart:128]: mmh, I was wrong. I use setter... Moved to public field (i just discovered that pair getter+setter is marked as unnecessary wrapper warning in intellij )

**Comment by ludoo0d0a** [comment]
> One enhancement comes to my mind: Remove duplicate values: If, for example, there's a word used in two searchstring files (maybe COMMON and ENGLISH or theres the same word in two languages), you don't need to add both... So the best would be to collect all string in a Set (=> removes duplicates) and convert this to a List afterwards

removeDupilcates added

File: lib/utils/common_utils.dart [hunk 4]
```diff
@@ -88,6 +88,12 @@ String normalizeUmlauts(String input) {
   }).join();
 }
 
+List<String> splitWords(String text){
```
[the remark this answers is not in the document]
Comment (ludoo0d0a) [reply lib/utils/common_utils.dart:91]: splitWords moved to Registry.splitWordsOnSpace

File: lib/widgets/main_view.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (ludoo0d0a) [reply lib/widgets/main_view.dart]: removed

**Pull request merged**